### PR TITLE
update: add support esm

### PR DESCRIPTION
**Pull request by RafaelGSS**
Hello!

I don't found in any doc the support at ESM on `fastify-autoload`. So I created it.
However, I don't have an idea of how to test it and make the support in previous versions. Any idea?

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify-autoload.d.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export interface AutoloadPluginOptions {
   dir: string
   ignorePattern?: RegExp | string
   includeTypeScript?: boolean
+  esmModules?: boolean
```
Comment (jsumners) [inline fastify-autoload.d.ts:10]: This is redundant. “ESM” === “ECMA Script Modules”. So this code is saying “ECMA Script Modules Modules”.

Comment (RafaelGSS) [reply fastify-autoload.d.ts:10]: `esModules` sounds good for you?

Comment (jsumners) [reply fastify-autoload.d.ts:10]: `loadESM: true`?

**Comment by RafaelGSS** [comment]
This PR is overridden by #76 ?

**Comment by jsumners** [comment]
I think so? I don't know. Please coordinate with the author in that thread.

**Pull request closed without merging**